In [2]:
report_card_dictionary = {"student_names": ["Vanshika", "Vivaan", "Siddhant", "Mitansh", "Kareena"],
                         "physics_grades": [34, 45, 56, 78, 100],
                         "chemistry_grades": [100, 78, 0, 45, 98],
                         "math_grades": [56, 76, 34, 23, 97], 
                          "biology_grades": [45, 56, 34, 67, 78]
                         }
report_card_dictionary

{'student_names': ['Vanshika', 'Vivaan', 'Siddhant', 'Mitansh', 'Kareena'],
 'physics_grades': [34, 45, 56, 78, 100],
 'chemistry_grades': [100, 78, 0, 45, 98],
 'math_grades': [56, 76, 34, 23, 97],
 'biology_grades': [45, 56, 34, 67, 78]}

In [3]:
# Setup
import pandas as pd

In [7]:
# Making the report card dictionary into a dataframe
report_card_df = pd.DataFrame(report_card_dictionary)
report_card_df

student_names  physics_grades  chemistry_grades  math_grades  biology_grades
0      Vanshika              34               100           56              45
1        Vivaan              45                78           76              56
2      Siddhant              56                 0           34              34
3       Mitansh              78                45           23              67
4       Kareena             100                98           97              78

In [11]:
# First 3 rows in the report card dataframe
report_card_df.head(3)

student_names  physics_grades  chemistry_grades  math_grades  biology_grades
0      Vanshika              34               100           56              45
1        Vivaan              45                78           76              56
2      Siddhant              56                 0           34              34

In [12]:
# 3 Random Rows in the report card dataframe
report_card_df.sample(3)

student_names  physics_grades  chemistry_grades  math_grades  biology_grades
2      Siddhant              56                 0           34              34
3       Mitansh              78                45           23              67
0      Vanshika              34               100           56              45

In [13]:
# Describing all the numeric columns in the report card dataframe
report_card_df.describe()

physics_grades  chemistry_grades  math_grades  biology_grades
count        5.000000          5.000000      5.00000        5.000000
mean        62.600000         64.200000     57.20000       56.000000
std         26.491508         42.144988     30.19437       17.392527
min         34.000000          0.000000     23.00000       34.000000
25%         45.000000         45.000000     34.00000       45.000000
50%         56.000000         78.000000     56.00000       56.000000
75%         78.000000         98.000000     76.00000       67.000000
max        100.000000        100.000000     97.00000       78.000000

# Insights

- What is the average chemistry grade? 64.2
- What is the average biology grade? 56.0
- What is the lowest math grade? 23.0
- What is the highest physics grade? 100
- How many scores are present in the math column? 5

In [25]:
# Make a total grade column
report_card_df["total_grade"] = report_card_df["physics_grades"] +report_card_df["math_grades"] + report_card_df["chemistry_grades"]+report_card_df["biology_grades"]

#Printed the report card dataframe to see the added column
report_card_df

student_names  physics_grades  chemistry_grades  math_grades  \
0      Vanshika              34               100           56   
1        Vivaan              45                78           76   
2      Siddhant              56                 0           34   
3       Mitansh              78                45           23   
4       Kareena             100                98           97   

   biology_grades  total_grade  
0              45          235  
1              56          255  
2              34          124  
3              67          213  
4              78          373

# Challenge: Make a column called average grade

In [27]:
report_card_df["average_grade"] = report_card_df["total_grade"]/4
report_card_df

student_names  physics_grades  chemistry_grades  math_grades  \
0      Vanshika              34               100           56   
1        Vivaan              45                78           76   
2      Siddhant              56                 0           34   
3       Mitansh              78                45           23   
4       Kareena             100                98           97   

   biology_grades  total_grade  average_grade  
0              45          235          58.75  
1              56          255          63.75  
2              34          124          31.00  
3              67          213          53.25  
4              78          373          93.25

In [28]:
report_card_df.describe()

physics_grades  chemistry_grades  math_grades  biology_grades  \
count        5.000000          5.000000      5.00000        5.000000   
mean        62.600000         64.200000     57.20000       56.000000   
std         26.491508         42.144988     30.19437       17.392527   
min         34.000000          0.000000     23.00000       34.000000   
25%         45.000000         45.000000     34.00000       45.000000   
50%         56.000000         78.000000     56.00000       56.000000   
75%         78.000000         98.000000     76.00000       67.000000   
max        100.000000        100.000000     97.00000       78.000000   

       total_grade  average_grade  
count     5.000000       5.000000  
mean    240.000000      60.000000  
std      89.615847      22.403962  
min     124.000000      31.000000  
25%     213.000000      53.250000  
50%     235.000000      58.750000  
75%     255.000000      63.750000  
max     373.000000      93.250000

In [32]:
# Saving the Report Card Insights Dataframe
report_card_insights_df = report_card_df.describe().transpose()
report_card_insights_df

count   mean        std    min     25%     50%     75%  \
physics_grades      5.0   62.6  26.491508   34.0   45.00   56.00   78.00   
chemistry_grades    5.0   64.2  42.144988    0.0   45.00   78.00   98.00   
math_grades         5.0   57.2  30.194370   23.0   34.00   56.00   76.00   
biology_grades      5.0   56.0  17.392527   34.0   45.00   56.00   67.00   
total_grade         5.0  240.0  89.615847  124.0  213.00  235.00  255.00   
average_grade       5.0   60.0  22.403962   31.0   53.25   58.75   63.75   

                     max  
physics_grades    100.00  
chemistry_grades  100.00  
math_grades        97.00  
biology_grades     78.00  
total_grade       373.00  
average_grade      93.25

In [35]:
# Saving the Report Card Insights as a CSV File
report_card_insights_df.to_csv("Report Card Insights New.csv")

# Report Card Alphabet Grades

In [36]:
report_card_df

student_names  physics_grades  chemistry_grades  math_grades  \
0      Vanshika              34               100           56   
1        Vivaan              45                78           76   
2      Siddhant              56                 0           34   
3       Mitansh              78                45           23   
4       Kareena             100                98           97   

   biology_grades  total_grade  average_grade  
0              45          235          58.75  
1              56          255          63.75  
2              34          124          31.00  
3              67          213          53.25  
4              78          373          93.25

In [40]:
# If Total Grade is greater than or equal to 140 make the Status Column Value as Pass 
report_card_df.loc[report_card_df["total_grade"]>=140, "Status"] = "Pass"

# If Total Grade is lower than 140 make the Status Column Value as Fail
report_card_df.loc[report_card_df["total_grade"]<140, "Status"] = "Fail"

# Print the Report Card DataFrame
report_card_df

student_names  physics_grades  chemistry_grades  math_grades  \
0      Vanshika              34               100           56   
1        Vivaan              45                78           76   
2      Siddhant              56                 0           34   
3       Mitansh              78                45           23   
4       Kareena             100                98           97   

   biology_grades  total_grade  average_grade Status  
0              45          235          58.75   Pass  
1              56          255          63.75   Pass  
2              34          124          31.00   Fail  
3              67          213          53.25   Pass  
4              78          373          93.25   Pass

In [44]:
# Making an Alphabet Grade Column
report_card_df.loc[(report_card_df["average_grade"] >=90), "Character_Grade"] = "A"
report_card_df.loc[(report_card_df["average_grade"] >=70) & (report_card_df["average_grade"]<90), "Character_Grade"] = "B"
report_card_df.loc[(report_card_df["average_grade"] >=50) & (report_card_df["average_grade"]<70), "Character_Grade"] = "C"
report_card_df.loc[(report_card_df["average_grade"] >=35) & (report_card_df["average_grade"]<50), "Character_Grade"] = "D"
report_card_df.loc[(report_card_df["average_grade"]<35), "Character_Grade"] = "F"

In [52]:
report_card_df

student_names  physics_grades  chemistry_grades  math_grades  \
0      Vanshika              34               100           56   
1        Vivaan              45                78           76   
2      Siddhant              56                 0           34   
3       Mitansh              78                45           23   
4       Kareena             100                98           97   

   biology_grades  total_grade  average_grade Status Character_Grade  
0              45          235          58.75   Pass               C  
1              56          255          63.75   Pass               C  
2              34          124          31.00   Fail               F  
3              67          213          53.25   Pass               C  
4              78          373          93.25   Pass               A

In [80]:
report_card_df = report_card_df.rename(columns={"Character_Grade": "Grades","average_grade": "Average Grades","total_grade": "Total Grade","biology_grades": "Biology Grades","math_grades": "Math Grades", "chemistry_grades": "Chemistry Grades","physics_grades": "Physics Grades", "student_names": "Student Names"})
report_card_df

Student Names  Physics Grades  Chemistry Grades  Math Grades  \
0      Vanshika              34               100           56   
1        Vivaan              45                78           76   
2      Siddhant              56                 0           34   
3       Mitansh              78                45           23   
4       Kareena             100                98           97   

   Biology Grades  Total Grade  Average Grades Status Grades  
0              45          235           58.75   Pass      C  
1              56          255           63.75   Pass      C  
2              34          124           31.00   Fail      F  
3              67          213           53.25   Pass      C  
4              78          373           93.25   Pass      A

In [81]:
report_card_df.to_csv("Report Card.csv")